crt_portal/cts_forms/filters.py

```diff
@@ -131,7 +131,11 @@ def report_filter(querydict):
                 kwargs[field] = querydict.getlist(field)
             elif field_options == 'violation_summary':
```

Comment:
It feels like this would be a good place for a switch statement - unrelated to this PR but maybe something we can refactor separately

Reply:
There's some fun python history behind this - Python didn't introduce a switch until 3.10, and it's not really a switch - it's pattern matching that [can be very complex if you feel like giving your reviewer a headache](https://towardsdatascience.com/the-match-case-in-python-3-10-is-not-that-simple-f65b350bb025).

https://stackoverflow.com/a/11479840/13231999

Plus, [the official docs](https://docs.python.org/3/tutorial/controlflow.html) still recommend elif:

<img width="810" alt="image" src="https://user-images.githubusercontent.com/15126660/218122440-88e5463b-54f0-4854-bf3f-ad012b799758.png">

Even in this case, it's important to note that the `if..elif` is more flexible than `match` when you're not comparing the _same value_ to several constants - for example, there's a condition (`if date in field`) in this example that would throw things off:

<img width="287" alt="image" src="https://user-images.githubusercontent.com/15126660/218122824-0baed8ad-9723-424b-b98b-49d7d1027631.png">

Reply:
all of that said, I agree, this could be refactored. Maybe into a helper function so we can return early and reduce nesting, e.g.:

```py
def build_filter(...):
  if field_options == 'etc':
    return qs.filter(...)
  if field_options == 'etc':
    return qs.filter(...)
  if field_options == 'etc':
    return qs.filter(...)  
  if field_options == 'etc':
    return qs.filter(...)


...


qs = build_filter(...)
```

Reply:
Interesting background. Makes sense - I wonder if we could just use `case create_date_start | create_date_end:` to solve that condition problem. I think it could be helpful for readability but that's just personal preference.